### PR TITLE
link-underline-removal

### PR DESCRIPTION
**Pull request by CarloBar1**
Task [AB#9329](https://dev.azure.com/neovici/Cosmoz3/_workitems/edit/9329/).
The underline on links should be shown only when the mouse hovers them.

The expected final behaviour is shown in the video:
https://user-images.githubusercontent.com/122988480/217528578-108df213-2dad-4ea3-9dba-e5895ddf03e7.mp4

### CODE REVIEW

**Comment by cristinecula** [comment]
Please include reference to issue.

File: cosmoz-omnitable-styles.js [hunk 1]
```diff
@@ -86,9 +86,11 @@ export default css`
 	//The following code makes the link underlining appear only on mouseover
 	:host a {
 		color: var(--primary-link-color, inherit);
+		text-decoration: none;
 	}
 	:host a:hover {
 		color: var(--primary-link-color, inherit);
+		color: lightblue;
```
Comment (cristinecula) [inline cosmoz-omnitable-styles.js:93]: Are you sure you want to override the color style here?

**Review by cristinecula — CHANGES_REQUESTED** [review]

File: cosmoz-omnitable-styles.js [hunk 2]
```diff
@@ -258,7 +267,6 @@ export default css`
 	.tableContent-empty.overlay div.tableContent-empty-message {
 		padding-bottom: 0;
 	}
-
```
Comment (cristinecula) [inline cosmoz-omnitable-styles.js:261]: SO many useless whitespace changes. Are they added by prettier? If not, please revert them.

File: cosmoz-omnitable-styles.js [hunk 3]
```diff
@@ -190,6 +195,10 @@ export default css`
 	cosmoz-omnitable-item-row > div[hidden] {
 		display: none !important;
 	}
+	cosmoz-omnitable-item-row > div a {
+		text-decoration: none;
+		color: grey;
```
Comment (cristinecula) [inline cosmoz-omnitable-styles.js:200]: If you are going to set a color, you should use a variable, to enable customization.

**Review by cristinecula — CHANGES_REQUESTED** [review]

File: cosmoz-omnitable-styles.js (hunk 3 above)
Comment (cristinecula) [reply cosmoz-omnitable-styles.js:200]: Is setting a color even necessary here?

File: cosmoz-omnitable-styles.js (hunk 2 above)
Comment (CarloBar1) [reply cosmoz-omnitable-styles.js:261]: I applied Prettier and it separates every selector's brackets with a whitespace - I just re-applied Prettier over the whole document now

File: cosmoz-omnitable-styles.js [hunk 4]
```diff
@@ -83,11 +83,15 @@ export default css`
 		position: relative;
 		overflow: hidden;
 	}
-
+	//The following code makes the link underlining appear only on mouseover
```
Comment (cristinecula) [inline cosmoz-omnitable-styles.js:86]: Comment is just explaining the obvious.
```suggestion
```

**Review by cristinecula — CHANGES_REQUESTED** [review]
I would like to have this configurable. Can we add `--cosmoz-omnitable-link-decoration` and `--cosmoz-omnitable-link-decoration--hover` variables?

File: cosmoz-omnitable-styles.js (hunk 4 above)
Comment (CarloBar1) [reply cosmoz-omnitable-styles.js:86]: Comment deleted, decoration variabilized

File: cosmoz-omnitable-styles.js [hunk 5]
```diff
@@ -190,6 +193,9 @@ export default css`
 	cosmoz-omnitable-item-row > div[hidden] {
 		display: none !important;
 	}
+	cosmoz-omnitable-item-row > div a {
+		text-decoration: none;
```
Comment (cristinecula) [inline cosmoz-omnitable-styles.js:197]: This should also use the css var.

File: cosmoz-omnitable-styles.js [hunk 6]
```diff
@@ -83,11 +83,14 @@ export default css`
 		position: relative;
 		overflow: hidden;
 	}
-
 	:host a {
 		color: var(--primary-link-color, inherit);
+		text-decoration: var(--cosmoz-omnitable-link-decoration, none);
```
Comment (cristinecula) [inline cosmoz-omnitable-styles.js:88]: Please keep the underline, as that is the default browser behaviour. We'll style it differently in frontend.
```suggestion
		text-decoration: var(--cosmoz-omnitable-link-decoration, underline);
```

**Review by cristinecula — CHANGES_REQUESTED** [review]

File: cosmoz-omnitable-styles.js [hunk 7]
```diff
@@ -190,6 +193,9 @@ export default css`
 	cosmoz-omnitable-item-row > div[hidden] {
 		display: none !important;
 	}
+	cosmoz-omnitable-item-row > div a {
+		--cosmoz-omnitable-link-decoration: none;
+	}
```
Comment (cristinecula) [inline cosmoz-omnitable-styles.js:198]:
```suggestion
```

This sets the `text-decoration` to none, which is the opposite of what we want.

**Review by cristinecula — CHANGES_REQUESTED** [review]

**Review by cristinecula — APPROVED** [review]

**Pull request merged**